Update sh_faction.lua

biorisk2 opened the pull request:
TO COMPETE WITH DARKRP, THIS IS ESSENTIAL.
SIMPLE JOBS MADE QUICKLY & WHILE THE SERVER IS RUNNING

Adding factions in nutscript is aids.
I wrote a shitty script 5 years ago to fix that, now I've written a new one that should just be in nutscript.

Registering factions can be done easily OUTSIDE of nutscript in a shared file post gamemode load with the following code, you can also add them while the server is running, or even generate them automagically with other code.

FACTION_TATOOINE = createNutJob(0, "Tatooine Citizen", Color(255,0,0), true, {"model/path/here.mdl"})

FACTION_CORUSCANT = createNutJob(1, "Coruscant Citizen", Color(255,0,0), false, {"model/path/here.mdl"})

FACTION_NARSHADDAA = createNutJob(2, "Nar Shaddaa Citizen", Color(255,0,0), false, {"model/path/here.mdl"})

biorisk2 committed:
Update sh_faction.lua

TO COMPETE WITH DARKRP, THIS IS ESSENTIAL.
SIMPLE JOBS MADE QUICKLY & WHILE THE SERVER IS RUNNING

Adding factions in nutscript is aids.
I wrote a shitty script 5 years ago to fix that, now I've written a new one that should just be in nutscript.

Registering factions can be done easily OUTSIDE of nutscript in a shared file post gamemode load with the following code, you can also add them while the server is running, or even generate them automagically with other code.

FACTION_TATOOINE = createNutJob(0, "Tatooine Citizen", Color(255,0,0), true, {"model/path/here.mdl"})

FACTION_CORUSCANT = createNutJob(1, "Coruscant Citizen", Color(255,0,0), false, {"model/path/here.mdl"})

FACTION_NARSHADDAA = createNutJob(2, "Nar Shaddaa Citizen", Color(255,0,0), false, {"model/path/here.mdl"})

DoopieWop commented:
ive never seen nutscript really as competing with darkrp. its competing more so with helix or clockwork. apart from that i dont think this is a necessary inclusion. I dont like the implementation of this either.

biorisk2 commented:
You don't like ease of access on creating factions to help server operators create their gamemodes faster & with ease?

What a weird stance bro.

biorisk2 committed:
Update sv_database.lua

factions that exceed 24 characters in length will silently fail on character creation, update, etc. because the SQL will quietly reject the change as it's limited to 24.

This allows for bigger file names & faction names.

FlorianLeChat commented on gamemode/core/libs/sh_faction.lua:
```diff
@@ -80,6 +80,31 @@ function nut.faction.getIndex(uniqueID)
 	return nut.faction.teams[uniqueID] and nut.faction.teams[uniqueID].index
 end
 
+function createNutJob(index, name, color, default, models)
```
This function shouldn't be included directly in `_G` but instead in the `nut.faction` namespace.

FlorianLeChat commented:
If this function is added to the gamemode, it should be called internally by the `nut.faction.loadFromDir` function to avoid code redundancy.

> TO COMPETE WITH DARKRP, THIS IS ESSENTIAL. SIMPLE JOBS MADE QUICKLY & WHILE THE SERVER IS RUNNING

I think NutScript has long since lost competition compared with other gamemodes (especially with DarkRP), but that doesn't prevent bringing new things and QOL improvements for servers refusing to migrate to another gamemode.

TovarischPootis commented:
I can see the value of an easy faction "maker" function. However, I agree with Florian that it should be within the `nut.faction` namespace.

Additionally, make this PR to the [1.2.3-wip branch](https://github.com/NutScript/NutScript/tree/1.2.3-wip) to prevent conficts.
